target/linux/ath79/dts/qca9533_yuncore_cpe830.dts

```diff
@@ -0,0 +1,162 @@
+                       color = <LED_COLOR_ID_GREEN>;
+                       gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+               };
+
+               lan {
+                       function = LED_FUNCTION_LAN;
+                       color = <LED_COLOR_ID_GREEN>;
+                       gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+               };
+
+               link1 {
+                       label = "green:link1";
+                       gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+               };
+
+               link2 {
+                       label = "green:link2";
+                       gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+               };
+
+               link3 {
+                       label = "green:link3";
+                       gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+               };
+
+               link4 {
+                       label = "green:link4";
+                       gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+               };
+       };
+};
+
+&spi {
+       status = "okay";
+
+       flash@0 {
+               compatible = "jedec,spi-nor";
+               reg = <0>;
+               spi-max-frequency = <25000000>;
+
+               partitions {
+                       compatible = "fixed-partitions";
+                       #address-cells = <1>;
+                       #size-cells = <1>;
+
+                       partition@0 {
+                               label = "u-boot";
+                               reg = <0x000000 0x040000>;
+                               read-only;
+                       };
+
+                       partition@40000 {
+                               label = "u-boot-env";
+                               reg = <0x040000 0x010000>;
+                       };
+
+                       partition@50000 {
+                               compatible = "denx,uimage";
+                               label = "firmware";
+                               reg = <0x050000 0xfa0000>;
+                       };
+
+                       partition@ff0000 {
+                               label = "art";
+                               reg = <0xff0000 0x010000>;
+                               read-only;
+
+                               nvmem-layout {
+					compatible = "fixed-layout";
+                                        #address-cells = <1>;
+                                        #size-cells = <1>;
+
+                                        macaddr_art_0: macaddr@0 {
+                                                reg = <0x0 0x6>;
+                                        };
+
+                                        macaddr_art_6: macaddr@6 {
+                                                reg = <0x6 0x6>;
+                                        };
+
+                                        cal_art_1000: calibration@1000 {
+                                                reg = <0x1000 0x440>;
+                                        };
+
+					macaddr_art_wmac: macaddr@1002 {
```

Comment:
This is part of the `cal_art_1000`. Is it needed to provide the `mac-address` separately if it is already part of the `calibration`? 

Please try to remove the `macaddr_art_wmac` and do not reference a mac address in the `wmac` and try if the wifi device still gets the correct mac address.

Reply:
Yes macaddr@1002 is the default MAC address offset. However, if we want to use the MAC address of a netdev as the label MAC, we must add the `"mac-address"` property.